hummingbot/connector/exchange/bitmax/bitmax_api_order_book_data_source.py

```diff
@@ -0,0 +1,260 @@
+                    f"HTTP status is {resp.status}."
+                )
+
+            resp_json = await resp.json()
+            if resp_json.get("code") != 0:
+                raise IOError(
+                    f"Error fetching last traded prices at {EXCHANGE_NAME}. "
+                    f"Error is {resp_json.message}."
+                )
+
+            for item in resp_json.get("data"):
+                trading_pair: str = convert_from_exchange_trading_pair(item.get("symbol"))
+                if trading_pair not in trading_pairs:
+                    continue
+
+                # we had to calculate mid price
+                ask = float(item.get("ask")[0])
+                bid = float(item.get("bid")[0])
+                result[trading_pair] = (ask + bid) / 2
+
+            return result
+
+    @staticmethod
+    async def fetch_trading_pairs() -> List[str]:
+        async with aiohttp.ClientSession() as client:
+            resp = await client.get(f"{REST_URL}/ticker")
+
+            if resp.status != 200:
+                # Do nothing if the request fails -- there will be no autocomplete for kucoin trading pairs
+                return []
+
+            data: Dict[str, Dict[str, Any]] = await resp.json()
+            return [convert_from_exchange_trading_pair(item["symbol"]) for item in data["data"]]
+
+    @staticmethod
+    async def get_order_book_data(trading_pair: str) -> Dict[str, any]:
+        """
+        Get whole orderbook
+        """
+        async with aiohttp.ClientSession() as client:
+            resp = await client.get(f"{REST_URL}/depth?symbol={convert_to_exchange_trading_pair(trading_pair)}")
+            if resp.status != 200:
+                raise IOError(
+                    f"Error fetching OrderBook for {trading_pair} at {EXCHANGE_NAME}. "
+                    f"HTTP status is {resp.status}."
+                )
+
+            data: List[Dict[str, Any]] = await safe_gather(resp.json())
+            item = data[0]
+            if item.get("code") != 0:
+                raise IOError(
+                    f"Error fetching OrderBook for {trading_pair} at {EXCHANGE_NAME}. "
+                    f"Error is {item.message}."
+                )
+
+            return item["data"]
+
+    async def get_new_order_book(self, trading_pair: str) -> OrderBook:
+        snapshot: Dict[str, Any] = await self.get_order_book_data(trading_pair)
+        snapshot_timestamp: float = snapshot.get("data").get("ts")
+        snapshot_msg: OrderBookMessage = BitmaxOrderBook.snapshot_message_from_exchange(
+            snapshot.get("data"),
+            snapshot_timestamp,
+            metadata={"trading_pair": trading_pair}
+        )
+        order_book = self.order_book_create_function()
+        active_order_tracker: BitmaxActiveOrderTracker = BitmaxActiveOrderTracker()
+        bids, asks = active_order_tracker.convert_snapshot_message_to_order_book_row(snapshot_msg)
```

Comment:
As discussed from our previous meeting, the `convert_snapshot_message_to_order_book_row` method is to be moved to `bitmax_utils.py`

Reply:
Thought we would stick with what we have similar to `crypto_com`?

Reply:
As close as we would like to follow `crypto_com` we are also looking to Cython files when integrating new connectors. In this case, removing the dependancy on `active_order_tracker` and moving the aforementioned method to a `utils` file would do